### PR TITLE
Add Brand type

### PR DESCRIPTION
**Pull request by airtoxin**
Fix https://github.com/piotrwitek/utility-types/issues/66

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -832,6 +833,29 @@ function makeStore(storeClass: Class<Store>): Store {
 
 [⇧ back to top](#flows-utility-types)
 
+### `Brand<T, U>`
```
Comment (piotrwitek) [inline README.md]: It should be located in Mapped Types

File: src/utility-types.ts [hunk 2]
```diff
@@ -79,3 +79,9 @@ export type $NonMaybeType<T> = NonNullable<T>;
  * @see https://flow.org/en/docs/types/utilities/#toc-class
  */
 export type Class<T> = new (...args: any[]) => T;
+
+/**
+ * Brand
+ * @desc Define nominal type of U based on type of T.
+ */
+export type Brand<T, U> = T & { __brand: U };
```
Comment (piotrwitek) [inline src/utility-types.ts]: please move to Mapped Types

File: src/utility-types.ts [hunk 3]
```diff
@@ -79,3 +79,9 @@ export type $NonMaybeType<T> = NonNullable<T>;
  * @see https://flow.org/en/docs/types/utilities/#toc-class
  */
 export type Class<T> = new (...args: any[]) => T;
+
+/**
+ * Brand
+ * @desc Define nominal type of U based on type of T.
+ */
```
Comment (piotrwitek) [inline src/utility-types.ts]: Add @example as here: https://github.com/piotrwitek/utility-types/pull/68

**Comment by airtoxin** [comment]
@piotrwitek Thank you reviewing! I fixed them. Please re-review 👓

File: src/mapped-types.ts [hunk 4]
```diff
@@ -245,3 +245,25 @@ export type WritableKeys<T extends object> = {
 export type ReadonlyKeys<T extends object> = {
   [P in keyof T]-?: IfEquals<{ [Q in P]: T[P] }, { -readonly [Q in P]: T[P] }, never, P>
 }[keyof T];
+
+/**
+ * Brand
+ * @desc Define nominal type of U based on type of T.
+ * @example
```
Comment (piotrwitek) [inline src/mapped-types.ts]: That's looking great.

**Comment by piotrwitek** [comment]
Thanks for your contribution @airtoxin!

**Pull request merged**